arkouda/suffix_array.py

```diff
@@ -0,0 +1,469 @@
+from __future__ import annotations
+
+from typing import cast, Union
+from typeguard import typechecked
+from arkouda.client import generic_msg
+from arkouda.pdarrayclass import pdarray, create_pdarray, _parse_single_int_array_value
+from arkouda.strings import Strings
+from arkouda.logger import getArkoudaLogger
+import numpy as np  # type: ignore
+from arkouda.dtypes import resolve_scalar_dtype, translate_np_dtype, int64
+import json
+
+__all__ = ["SArrays", "suffix_array", "lcp_array", "suffix_array_file"]
+
+
+class SArrays:
+    """
+    Represents an array of (suffix) arrays whose data resides on the arkouda server.
+    The user should not call this class directly; rather its instances are created
+    by other arkouda functions. It is very similar to Strings and the difference is
+    that its content is int arrays instead of strings.
+
+    Attributes
+    ----------
+    offsets : pdarray
+        The starting indices for each suffix array
+    bytes : pdarray
+        The raw integer indices of all suffix arrays
+    size : int
+        The number of suffix arrays in the array
+    nbytes : int
+        The total number of indices in all suffix arrays
+        We have the same number indices as the number of characters/suffixes in strings
+    ndim : int
+        The rank of the array (currently only rank 1 arrays supported)
+    shape : tuple
+        The sizes of each dimension of the array
+    dtype : dtype
+        The dtype is np.int
+    logger : ArkoudaLogger
+        Used for all logging operations
+
+    Notes
+    -----
+    SArrays is composed of two pdarrays: (1) offsets, which contains the
+    starting indices for each string's suffix array  and (2) bytes, which contains the
+    indices of all suffix arrays, no any spliter between two index arrays.
+    """
+
+    BinOps = frozenset(["==", "!="])
+    objtype = "int"
```

Comment:
I recommend replacing the objtype string with type argument or Enum

Reply:
That makes sense to me. I think this is just following along with what is done in `Strings` and `pdarrayclass` where the `objtype` is a literal string. Would it make sense to hold off on this change and update along with the others in a separate ticket?

https://github.com/Bears-R-Us/arkouda/blob/b83c3ba320ef7f64cc3521b487f4efbea88b162a/arkouda/pdarrayclass.py#L99

https://github.com/Bears-R-Us/arkouda/blob/b83c3ba320ef7f64cc3521b487f4efbea88b162a/arkouda/strings.py#L52

Reply:
@pierce314159 yeah, good idea to work that in a separate issue